src/Uno.Extensions.Navigation.UI/Navigators/MessageDialogNavigator.cs

```diff
@@ -95,7 +95,7 @@ protected override Task<bool> RegionCanNavigate(Route route, RouteInfo? routeMap
 		};
 		md.Commands.AddRange(commands);
 
-#if WINUI && WINDOWS
+#if WINUI
 		var window = Window!;
 		var hwnd = WinRT.Interop.WindowNative.GetWindowHandle(window);
 		WinRT.Interop.InitializeWithWindow.Initialize(md, hwnd);
```

Comment:
The condition here should likely be `#if WINUI || HAS_UNO_WINUI`. On targets with multiwindow support, `MessageDialog` also now requires this `InitializeWithWindow` snippet

Reply:
Updated